### PR TITLE
Recipients endpoint

### PR DESCRIPTION
**Pull request by AbdelilahLbardi**
Changelist

- [x] Add support for `GET` & `DELETE` to the `HttpLayer`.
- [x] `Recipient` list endpoint.
- [x] `Recipients` find endpoint.
- [x] `Recipients` delete endpoint.
- [x] Documentation update

Breaking changes:
- N/A

### CODE REVIEW

File: tests/Endpoints/RecipientTest.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        parent::setUp();
+
+        $this->client = new Client();
+
+        $this->recipients = new Recipient(new HttpLayer(self::OPTIONS, $this->client), self::OPTIONS);
+        $this->defaultResponse = $this->createMock(ResponseInterface::class);
+        $this->defaultResponse->method('getStatusCode')->willReturn(200);
+    }
+
+    public function test_get_messages_min_limit_is_validated()
```
Comment (fosron) [inline tests/Endpoints/RecipientTest.php:28]: `test_get_messages` -> `test_get_recipients`

File: tests/Endpoints/RecipientTest.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        $this->expectExceptionMessage('Minimum limit is ' . Recipient::MIN_LIMIT . '.');
+
+        $this->recipients->get(9);
+    }
+
+    public function test_get_messages_max_limit_is_validated()
```
Comment (fosron) [inline tests/Endpoints/RecipientTest.php:35]: `test_get_messages` -> `test_get_recipients`

File: tests/Endpoints/RecipientTest.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        $this->expectExceptionMessage('Maximum limit is ' . Recipient::MAX_LIMIT . '.');
+
+        $this->recipients->get(101);
+    }
+
+    public function test_get_messages()
```
Comment (fosron) [inline tests/Endpoints/RecipientTest.php:42]: `test_get_messages` -> `test_get_recipients`

**Review by fosron — CHANGES_REQUESTED** [review]

**Comment by fosron** [comment]
Also, in README examples we should have `$mailersend = new MailerSend(['api_key' => 'key']);`

**Comment by AbdelilahLbardi** [comment]
@aorfanos Branch updated!

**Review by fosron — APPROVED** [review]

File: src/Endpoints/Recipient.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        return $this->httpLayer->get(
+            $this->buildUri($this->endpoint),
+            array_filter([
+                'limit' => $limit,
+                'page' => $page
+            ])
+        );
+    }
```
Comment (doobas) [inline src/Endpoints/Recipient.php:41]: I'm missing `domain_id` parameter here

**Review by doobas — CHANGES_REQUESTED** [review]

File: src/Endpoints/Recipient.php (hunk 4 above)
Comment (fosron) [reply src/Endpoints/Recipient.php:41]: https://developers.mailersend.com/api/v1/recipients.html#get-recipients

Comment (doobas) [reply src/Endpoints/Recipient.php:41]: I have checked with the current app code. We have domain filter there. Maybe we should update our docs?

Comment (fosron) [reply src/Endpoints/Recipient.php:41]: yeah, its missing in the docs too, most likely will be missing in all SDKs :)

Comment (doobas) [reply src/Endpoints/Recipient.php:41]: whoops :)

Comment (AbdelilahLbardi) [reply src/Endpoints/Recipient.php:41]: @doobas I double checked this one, I will add `domain_id` optional param, and update out developers doc aswell.

Comment (fosron) [reply src/Endpoints/Recipient.php:41]: @AbdelilahLbardi i did update dev docs before

**Pull request merged**